packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -237,6 +237,12 @@ export const GridRootStyles = styled('div', {
         overflow: 'visible !important',
         whiteSpace: 'nowrap',
       },
+      [`& .${c.groupingCriteriaCell}`]: {
+        width: 'unset',
```

Comment:
It doesn't seem to work for nested groups, probably due to the margin-left offset applied to the nested cell content: https://codesandbox.io/p/sandbox/focused-thunder-pf5kdw?file=%2Fsrc%2FDemo.tsx%3A29%2C33

https://github.com/mui/mui-x/assets/13808724/ee1f0714-8a94-4de2-a628-473dfe32ec19

Should we change it to padding here?

https://github.com/mui/mui-x/blob/3a3263bef0ebeb52416b79a5b72fa431e2e50054/packages/x-data-grid-premium/src/components/GridGroupingCriteriaCell.tsx#L78-L83

Same for tree data.

Reply:
Yeah I think I'll rework the autosizing approach a bit with the cell DOM change, it will fix autosize for this case and the new cell DOM.